src/reanimated2/shareables.ts

```diff
@@ -329,8 +329,12 @@ export function makeShareableCloneOnUIRecursive<T>(
   return cloneRecursive(value);
 }
 
-export function makeShareable<T extends object>(value: T): T {
-  if (SHOULD_BE_USE_WEB) {
+function makeShareableJS<T extends object>(value: T): T {
+  return value;
+}
+
+function makeShareableNative<T extends object>(value: T): T {
+  if (shareableMappingCache.get(value)) {
```

Comment:
Adding simple optimization here so we won't call `makeShareableCloneRecursive` and `shareableMappingCache` multiple times for no effect. If for safety reason you don't want it to be in this PR, I can move it to separate PR.

Although if `value` was mapped to some other shareable already, then the previous worklet that was using it could get broken with current implementation.